### PR TITLE
Support configuring the set of codegen backends to build per host triple

### PR DESCRIPTION
**Pull request by bjorn3**
This allows building the compiler itself with one backend while using another backend at runtime. For example this allows compiling rustc to wasm using LLVM, while using Cranelift at runtime to produce actual code. Cranelift can't compile to wasm, but is perfectly capable of running on wasm. LLVM can compile to wasm, but can't run on wasm. [^1]

[^1]: The prototype of this still requires a couple of other patches.

### CODE REVIEW

File: src/bootstrap/src/core/config/config.rs [hunk 1]
```diff
@@ -581,6 +581,7 @@ pub struct Target {
     pub wasi_root: Option<PathBuf>,
     pub qemu_rootfs: Option<PathBuf>,
     pub no_std: bool,
+    pub codegen_backends: Option<Vec<Interned<String>>>,
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/config/config.rs:584]: Can you add a change info for this field? Similar to https://github.com/rust-lang/rust/blob/04521fd10e6eefb113ca1cefd02862599fd08640/src/bootstrap/src/utils/change_tracker.rs#L74-L78

Comment (bjorn3) [reply src/bootstrap/src/core/config/config.rs:584]: Done

**Comment by albertlarsan68** [comment]
How does this interact with stages ?

**Comment by bjorn3** [comment]
If the host triple matches the build triple, then setting it in `target.<triple>.codegen-backends` will apply to all stages. If they don't match, it will apply to rustc compiled for the respective target.

**Comment by albertlarsan68** [comment]
Thanks for the PR!
@bors r+

**Pull request merged**